src/datasets/datasets.controller.ts

```diff
@@ -409,16 +409,16 @@ export class DatasetsController {
       version: "v3",
     };
 
+    if ("proposalId" in inputObsoleteDataset) {
+      propertiesModifier.proposalIds = [
+        (inputObsoleteDataset as CreateRawDatasetObsoleteDto).proposalId,
+      ];
+    }
```

Comment:
Is it intentional that it's been scoped out of the if condition below? instead of putting together with sampleId and instrumentId conversions?

Reply:
Yes, as I added proposal id also to derived datasets